pennylane/devices/device_api.py

```diff
@@ -0,0 +1,447 @@
+          has its own set of configuration parameters. For example, every script can be executed with a different
+          number of shots.
+
+        **Interface parameters:**
+
+        Note that the parameters contained within the quantum script may contain interface-specific data types, such as
+        ``torch.tensor`` or ``tensorflow.Variable``. If the device does not wish to handle interface-specific parameters, they
+        can implement an optional "internal preprocessing" step that converts all parameters to vanilla numpy. A convenience
+        transform implementing this will be provided.
+
+        **Return shape:**
+
+        The result for each :class:`~.QuantumScript` must match the shape specified by :class:`~.QuantumScript.shape`.
+
+        The level of priority for dimensions from outer dimension to inner dimension is:
+
+        1) Quantum Script in batch
+        2) Shot choice in a shot vector
+        3) Measurement in the quantum script
+        4) Parameter broadcasting
+        5) Measurement shape for array-valued measurements like probabilities
+
+        For a batch of quantum scripts with multiple measurements, a shot vector, and parameter broadcasting:
+
+        * ``result[0]``: the results for the first script
+        * ``result[0][0]``: the first shot number in the shot vector
+        * ``result[0][0][0]``: the first measurement in the quantum script
+        * ``result[0][0][0][0]``: the first parameter broadcasting choice
+        * ``result[0][0][0][0][0]``: the first value for an array-valued measurement
+
+        With the exception of quantum script batches, dimensions with only a single component should be eliminated.
+
+        For example:
+
+        With a single script and a single measurement process, execute should return just the
+        measurement value in a numpy array. ``shape`` currently accepts a device, as historically devices
+        stored shot information. In the future, this method will accept an ``ExecutionConfig`` instead.
+
+        >>> qs = qml.tape.QuantumScript([], [qml.expval(qml.PauliZ(0))])
+        >>> qs.shape(dev)
+        ()
+        >>> dev.execute(qs)
+        array(1.0)
+
+        If execute recieves a batch of scripts, then it should return a tuple of results:
+
+        >>> dev.execute([qs, qs])
+        (array(1.0), array(1.0))
+        >>> dev.execute([qs])
+        (array(1.0),)
+
+        If the script has multiple measurments, then the device should return a tuple of measurements.
+
+        >>> qs = qml.tape.QuantumScript([], [qml.expval(qml.PauliZ(0)), qml.probs(wires=(0,1))])
+        >>> qs.shape(dev)
+        ((), (4,))
+        >>> dev.execute(qs)
+        (array(1.0), array([1., 0., 0., 0.]))
+
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def supports_jacobian_with_configuration(cls, execution_config=None) -> bool:
```

Comment:
@albi3ro the `with_configuration` here seems superfluous to me, as the execution config is the argument.

So perhaps it should just be `supports_gradient`?

Reply:
On another note, the docstring uses `supports_gradient` which does not agree with the actual method name above